src/features/QueryStatus.tsx

```diff
@@ -0,0 +1,53 @@
+import { UseQueryResult } from "@tanstack/react-query";
+import { Children, ReactNode } from "preact/compat";
+
+type Props = {
+  children: ReactNode;
+
+  /** Tanstack Query object */
+  query: UseQueryResult<unknown, Error>;
+
+  /** Given query data is considered 'empty' when method is true */
+  isEmpty: (arg: any) => boolean;
+};
+
+export const Loading = ({ children }) => <>{children}</>;
+
+export const Error = ({ children }) => <>{children}</>;
+
+export const Empty = ({ children }) => <>{children}</>;
+
+export const Success = ({ children }) => <>{children}</>;
+
+/**
+ * Given a Tanstack Query and children for each state, render the child matching
+ * the current state of the query
+ */
+export const QueryStatus = ({ children, isEmpty, query }: Props) => {
+  let content = null;
+
+  // Loop through children and find the matching status component
+  Children.forEach(children, (child) => {
+    // @ts-expect-error: children have a type prop with the component object
```

Comment:
I fought with the children prop type for a bit. I tried a few different things like ComponentChild, but nothing had this 'type' prop which is literally the component object. I even tried extending a new type or interface. Gave up. This is fine.